Lib/ast.py

```diff
@@ -1273,10 +1276,13 @@ def visit_IfExp(self, node):
             self.traverse(node.orelse)
 
     def visit_Set(self, node):
-        if not node.elts:
-            raise ValueError("Set node should have at least one item")
-        with self.delimit("{", "}"):
-            self.interleave(lambda: self.write(", "), self.traverse, node.elts)
+        if node.elts:
+            with self.delimit("{", "}"):
+                self.interleave(lambda: self.write(", "), self.traverse, node.elts)
+        else:
+            # `{}` would be interpreted as a dictionary literal, and
+            # `set` might not be shadowed. Thus:
+            self.write('{1}.__class__()')
```

Comment:
Nice!

Reply:
Perhaps `{*()}` would be a bit more concise?

Reply:
I agree, the `{1}` can be a bit confusing.

Reply:
just for the reference, the proposed was is how astor does it. Though I think `{*()}` is also a good alternative

Reply:
Yep, I'm the one who wrote that astor patch. I didn't realize `{*()}` was a possibility.

Reply:
< joking>

What about this nuclear option?

```
>>> ctypes.pythonapi.PySet_New.argtypes = ctypes.py_object,
>>> ctypes.pythonapi.PySet_New.restype = ctypes.py_object
ctypes.pythonapi.PySet_New([]).__class__
```

< /joking>